### PR TITLE
Add CallToAction component

### PR DESCRIPTION
**Pull request by jxnblk**
image

### CODE REVIEW

File: gatsby-theme-landing-page/src/components/button.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import * as React from 'react'
+import { Link } from "gatsby";
+import * as styles from './button.module.css'
```
Comment (jxnblk) [inline gatsby-theme-landing-page/src/components/button.js]: Looks like we don't have prettier set up for commits; can look into that later

File: src/components/layout.module.css [hunk 2]
```diff
@@ -8,6 +8,7 @@
 }
 
 .header {
+  composes: container from "gatsby-theme-landing-page/src/components/base.module.css";
```
Comment (jxnblk) [inline src/components/layout.module.css]: Just testing to see if this works with shadowing; might pull global base styles back out to the starter and keep these more decoupled

**Comment by jxnblk** [comment]
I added lint-staged and prettier to this PR as well, so the change looks bigger than it really is

File: gatsby-theme-landing-page/src/components/button.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import * as React from "react";
+import { Link } from "gatsby";
+import * as styles from "./button.module.css";
+
+export default function Button({ href, text, children, variant = "primary" }) {
```
Comment (thinkybeast) [inline gatsby-theme-landing-page/src/components/button.js:5]: Cool - not necessary for this PR, but I'm thinking we pull these "base" components out to their own directory e.g. `src/components/base`, to differentiate them from the Contentful section components. WDYT?

**Review by thinkybeast — APPROVED** [review]
Looks great

File: gatsby-theme-landing-page/src/components/button.js (hunk 3 above)
Comment (jxnblk) [reply gatsby-theme-landing-page/src/components/button.js:5]: I was thinking something similar, but with the top-level components. I'm guessing we'll have some other shared components to pull out, so definitely prefer to wait until we have the basics working, then take an audit of what we have and organize as needed. The more directories we create, the more complex that makes the API for end-users of the theme

**Pull request merged**